contracts/vault/StrategySplitterV2.sol

```diff
@@ -348,37 +349,31 @@ contract StrategySplitterV2 is ControllableV3, ReentrancyGuard, ISplitter {
     }
 
     // if we are able to withdraw something let's do it
+    uint strategyLossOnWithdraw;
     if (lowStrategyBalance != 0) {
-      int totalAssetsDelta = (percent == 100)
+      strategyLossOnWithdraw = (percent == 100)
       ? IStrategyV2(lowStrategy).withdrawAllToSplitter()
       : IStrategyV2(lowStrategy).withdrawToSplitter(lowStrategyBalance * percent / 100);
-      if (totalAssetsDelta != 0) {
-        balance = _fixTotalAssets(balance, totalAssetsDelta);
-      }
     }
-    uint balanceAfterWithdraw = totalAssets();
 
-    (address topStrategy,) = _investToTopStrategy(
+    (address topStrategy, uint strategyLossOnInvest) = _investToTopStrategy(
       false // we assume here, that total-assets-amount of the strategy was just updated in withdraw above
     );
     require(topStrategy != address(0), "SS: Not invested");
 
-    uint balanceAfterInvest = totalAssets();
     uint slippage;
     // slippage for withdraw
-    if (balanceAfterWithdraw < balance) {
-      uint loss = balance - balanceAfterWithdraw;
-      ITetuVaultV2(vault).coverLoss(loss);
-      emit Loss(lowStrategy, loss);
-      slippage = loss * 100_000 / balance;
+    if (strategyLossOnWithdraw != 0) {
+      ITetuVaultV2(vault).coverLoss(strategyLossOnWithdraw);
+      emit Loss(lowStrategy, strategyLossOnWithdraw);
+      slippage = strategyLossOnWithdraw * 100_000 / balance;
```

Comment:
Now strategyLossOnWithdraw can include loss related to price-changes. So, in the case of high price change you will get high value of loss and most probably too-high slippage. The slippage shouldn't depend on price changes (?) There is no such problem with strategyLossOnInvest below because strategyLossOnInvest doesn't include losses related to price changes.

Reply:
its fine
probably for invest need to check too